Core/include/Acts/TrackFinding/MeasurementSelector.hpp

```diff
@@ -168,30 +190,16 @@ class MeasurementSelector {
     const auto numMeasurementsCut = VariableCut<traj_t>(
         *candidates.begin(), cuts, cuts->numMeasurementsCutOff, logger);
 
-    auto endIterator = candidates.begin();
-    auto maxIterator = candidates.end();
-    if (candidates.size() > numMeasurementsCut && numMeasurementsCut > 0) {
-      maxIterator = std::next(candidates.begin(), numMeasurementsCut);
+    if ( static_cast<std::size_t>(std::distance(candidates.begin(),trackStateIterEnd)) > numMeasurementsCut && numMeasurementsCut > 0) {
+       trackStateIterEnd = std::next(candidates.begin(), numMeasurementsCut);
```

Comment:
Minor point and irrelevant for `std::vector`: if `numMeasurementsCut > 0` comes first, the call to `std::distance` can be avoided if that's not the case.

Actually: if `numMeasurementsCut` is 0 we can probably early return from the function anyway?

Reply:
If distance was an expensive operation this would be worthwhile, but likely distance is cheaper than the comparison. 
This is mustly just:
`return std::next(candidates.begin(), min( numMeasurementsCut,s_c<>( std::distance(candidates.begin(),trackStateIterEnd))) );`
one cannot return much earlier.

Reply:
`std::distance` is O(1) for `vector`, but not for other containers. But yeah, in this case it does not matter, really.